dbms/src/Storages/DeltaMerge/SegmentReadTaskPool.cpp

```diff
@@ -98,39 +132,9 @@ SegmentReadTasks SegmentReadTask::trySplitReadTasks(const SegmentReadTasks & tas
 BlockInputStreamPtr SegmentReadTaskPool::buildInputStream(SegmentReadTaskPtr & t)
 {
     MemoryTrackerSetter setter(true, mem_tracker.get());
-    auto seg = t->segment;
     BlockInputStreamPtr stream;
-    auto block_size = std::max(expected_block_size, static_cast<size_t>(dm_context->db_context.getSettingsRef().dt_segment_stable_pack_rows));
-    switch (read_mode)
-    {
-    case ReadMode::Normal:
-        stream = seg->getInputStream(
-            *dm_context,
-            columns_to_read,
-            t->read_snapshot,
-            t->ranges,
-            filter,
-            max_version,
-            block_size);
-        break;
-    case ReadMode::Fast:
-        stream = seg->getInputStreamFast(
-            *dm_context,
-            columns_to_read,
-            t->read_snapshot,
-            t->ranges,
-            filter,
-            block_size);
-        break;
-    case ReadMode::Raw:
-        stream = seg->getInputStreamRaw(
-            *dm_context,
-            columns_to_read,
-            t->read_snapshot,
-            t->ranges);
-        break;
-    }
-    LOG_FMT_DEBUG(log, "getInputStream succ, pool_id={} segment_id={}", pool_id, seg->segmentId());
+    buildStreamBasedOnReadMode(stream, read_mode, t, dm_context, columns_to_read, filter, max_version, expected_block_size);
```

Comment:
How about this?

- getInputStream(Mode, ...)
- getInputStreamModeNormal(...)
- getInputStreamModeFast(...)
- getInputStreamModeRaw(...)

Reply:
Good suggestion! I think getInputStream(Mode, ...) in class Segment is a better way to make the code reuseable and clean.